src/main/java/com/amazon/opendistroforelasticsearch/security/privileges/PrivilegesEvaluator.java

```diff
@@ -353,6 +354,35 @@ public PrivilegesEvaluatorResponse evaluate(final User user, String action0, fin
         final Set<String> allIndexPermsRequired = evaluateAdditionalIndexPermissions(request, action0);
         final String[] allIndexPermsRequiredA = allIndexPermsRequired.toArray(new String[0]);
 
+        // handle resolve requests
+        if (action0.equals("indices:admin/resolve/index")) {
+
+            if(requestedResolved.getAllIndices().isEmpty()) {
```

Comment:
Is it possible to short circuit the condition for `do_not_fail_on_forbidden` line `#389` and reuse the same logic ?

Reply:
@sujithvm do you mean refactor code under `do_not_fail_on_forbidden` condition? I am not sure what do you mean by short circuit?

Reply:
By [short circuit](https://en.wikipedia.org/wiki/Short-circuit_evaluation) here I mean, add the `action0.equals("indices:admin/resolve/index")` to the if statement as it currently executes only if `do_not_fail_on_forbidden` is enabled. 

Refactoring the code to reuse it again would be fine as well

Reply:
I don't think that combining `do_not_fail_on_forbidden` with other conditions is the right approach.